sample-3.md

```diff
@@ -0,0 +1,536 @@
+event consumers, so that we can show how to selectively send events to distinct
+event consumers later.
+
+Setup the '-dev' `Service` and `Deployment`.
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: event-display-dev
+spec:
+  replicas: 1
+  selector:
+    matchLabels: &labels
+      app: event-display-dev
+  template:
+    metadata:
+      labels: *labels
+    spec:
+      containers:
+        - name: event-display
+          # Source code: https://github.com/knative/eventing-sources/blob/release-0.6/cmd/event_display/main.go
+          image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display@sha256:37ace92b63fc516ad4c8331b6b3b2d84e4ab2d8ba898e387c0b6f68f0e3081c4
+
+---
+
+# Service pointing at the previous Deployment. This will be the target for event
+# consumption.
+kind: Service
+apiVersion: v1
+metadata:
+  name: event-display-dev
+spec:
+  selector:
+    app: event-display-dev
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+END
+```
+
+Now setup the '-test' `Service` and `Deployment`:
+
+```shell
+cat << END | kubectl -n $K_NAMESPACE apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: event-display-test
+spec:
+  replicas: 1
+  selector:
+    matchLabels: &labels
+      app: event-display-test
+  template:
+    metadata:
+      labels: *labels
+    spec:
+      containers:
+        - name: event-display
+          # Source code: https://github.com/knative/eventing-sources/blob/release-0.6/cmd/event_display/main.go
+          image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display@sha256:37ace92b63fc516ad4c8331b6b3b2d84e4ab2d8ba898e387c0b6f68f0e3081c4
+
+---
+
+# Service pointing at the previous Deployment. This will be the target for event
+# consumption.
+kind: Service
+apiVersion: v1
+metadata:
+  name: event-display-test
+spec:
+  selector:
+    app: event-display-test
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8080
+END
+```
+
+Let's wait for the `Deployment`s to be ready, as shown by their Available status
+condition.
+
+```shell
+kubectl -n $K_NAMESPACE get deployments event-display-dev event-display-test                                                                                                                                            +54 10:30 ❰─┘
+```
+
+This should return something like:
+```shell
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+event-display-dev    1         1         1            1           16m
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+event-display-test   1         1         1            1           16m
+```
+
+In particular, we want the number in the `DESIRED` column to match the number in the
+`AVAILABLE` column. This may take a few minutes. If after three minutes, it the
+numbers still do not match, then investigate why. 
+
+TODO Add an investigation guide.
```

Comment:
Should we add a troubleshooting section and then as needed add more sub pages to it? Could you put this Todos in the document - title and brief description of what the content should be. Someone else can pick this too.

Reply:
Absolutely, we should add a set of troubleshooting guides. 

This in particular will be a very simple guide, as it is a standard K8s Deployment, albeit with a Knative image.